src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -339,13 +339,31 @@ private static function dnsResolve(string $host, NativeClientState $multi, array
             $info['debug'] .= "* Hostname was NOT found in DNS cache\n";
             $now = microtime(true);
 
-            if (!$ip = gethostbynamel($host)) {
+            if ($ip = gethostbynamel($host)) {
+                $multi->dnsCache[$host] = $ip = $ip[0];
+                $info['debug'] .= "* Added {$host}:0:{$ip} to DNS cache\n";
+                $host = $ip;
+            } elseif (\defined('STREAM_PF_INET6')) {
+                if ($ip = dns_get_record($host, \DNS_AAAA)) {
+                    $multi->dnsCache[$host] = $ip = $ip[0]['ipv6'];
+                    $info['debug'] .= "* Added {$host}:0:{$ip} to DNS cache\n";
```

Comment:
these lines are missing corresponding ones in the sockets path.
I think they should be kept outside the if like before, with some adaptations of course